Update kubeadm tests

detiber opened the pull request:
- Add missing kubeadm 1.9 tests
- Add kubeadm 1.10 tests

Signed-off-by: Jason DeTiberus <detiber@gmail.com>

Additional configuration will be needed in coordination with https://github.com/kubernetes/test-infra/pull/6834

detiber commented:
@timothysc ptal

detiber commented:
Second commit in this series currently removes kubeadm 1.10 tests from the config due to the missing prow jobs for 1.10 builds.

BenTheElder commented on prow/config.yaml:
```diff
@@ -5634,6 +5634,30 @@ postsubmits:
+          args:
+          - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+          - "--clean"
+          - "--git-cache=/root/.cache/git"
```
drop this flag and the volume / mount / host port, we're trying to eliminate this and normally most CI tests don't use this

BenTheElder replied:
*please* :-), I'll cleanup some more of the other jobs in a follow-up

BenTheElder commented:
Thanks! one comment, also noting that ~ tomorrow I'll be taking a pass at improving kubeadm CI, got backlogged with a few other things but I want to get this healthier.

detiber commented:
@BenTheElder I've rebased on top of master since https://github.com/kubernetes/test-infra/pull/6834 was merged. I believe that I've also addressed the changes that you requested.
